pymc3/distributions/distribution.py

```diff
@@ -56,17 +56,32 @@ def __new__(cls, name, *args, **kwargs):
                             "a 'with model:' block, or use the '.dist' syntax "
                             "for a standalone distribution.")
 
-        if isinstance(name, string_types):
-            data = kwargs.pop('observed', None)
-            cls.data = data
-            if isinstance(data, ObservedRV) or isinstance(data, FreeRV):
-                raise TypeError("observed needs to be data but got: {}".format(type(data)))
-            total_size = kwargs.pop('total_size', None)
-            dist = cls.dist(*args, **kwargs)
-            return model.Var(name, dist, data, total_size)
-        else:
+        if not isinstance(name, string_types):
             raise TypeError("Name needs to be a string but got: {}".format(name))
 
+        data = kwargs.pop('observed', None)
+        cls.data = data
+        if isinstance(data, ObservedRV) or isinstance(data, FreeRV):
+            raise TypeError("observed needs to be data but got: {}".format(type(data)))
+        total_size = kwargs.pop('total_size', None)
+
+        dims = kwargs.pop('dims', None)
+        has_shape = 'shape' in kwargs
+        shape = kwargs.pop('shape', None)
```

Comment:
These two lines seem redundant

Reply:
I think there are not. I'm making sure that `shape` will not be in kwargs if it wasn't specified.
Some distributions are somewhat peculiar in how they react to `shape=None` vs no shape arg at all. (last time I checked anyway)

Reply:
Ooh, I see it now! Didn't know about that, thanks for the explanation Adrian 👌